### PR TITLE
Add a few small IO related and a ground_state global

### PR DESCRIPTION
**Pull request by marius851000**
A few small stuff. Not sure how large I should try to make my PR (if any).

### CODE REVIEW

**Comment by marius851000** [comment]
Do you know how I should name "FreeIOVec" and "IsIOVecAllocated" ?

File: symbols/arm9.yml [hunk 1]
```diff
@@ -626,6 +634,13 @@ arm9:
         Note: It is possible to keep a file stream open even if data transfer mode has been stopped, in which case the file stream can be used again if data transfer mode is reinitialized.
         
         r0: file_stream pointer
+    - name: FreeIOVec
```
Comment (UsernameFodder) [inline symbols/arm9.yml:637]: `FreeIoVec`

File: symbols/arm9.yml [hunk 2]
```diff
@@ -344,6 +344,14 @@ arm9:
         See https://developer.arm.com/documentation/dui0474/k/image-structure-and-generation/linker-generated-veneers/what-is-a-veneer-
         
         No params.
+    - name: IsIOVecAllocated
```
Comment (UsernameFodder) [inline symbols/arm9.yml:347]: `IsIoVecAllocated`

File: headers/types/common/file_io.h [hunk 3]
```diff
@@ -140,4 +140,19 @@ struct wte_header {
 };
 ASSERT_SIZE(struct wte_header, 32);
 
+struct bg_list_entry_sub_struct {
+    char name[9]; // Zero terminated string
+    undefined1 field_0x9;
+};
+ASSERT_SIZE(struct bg_list_entry_sub_struct, 10);
+
+// Represent an entry in the bg_list.dat file, after being loaded by LoadABgListEntry
```
Comment (UsernameFodder) [inline headers/types/common/file_io.h:149]: `LoadBgListEntry` if we make that change.

File: headers/types/common/file_io.h [hunk 4]
```diff
@@ -140,4 +140,19 @@ struct wte_header {
 };
 ASSERT_SIZE(struct wte_header, 32);
 
+struct bg_list_entry_sub_struct {
+    char name[9]; // Zero terminated string
+    undefined1 field_0x9;
```
Comment (theCapypara) [inline headers/types/common/file_io.h:145]: This is probably just a null terminator? If so; Would it maybe make sense to just include it in name?

**Comment by UsernameFodder** [comment]
@marius851000 I made the suggested changes on this myself since they're fairly unambiguous and this PR has been sitting open for months. I also resolved merge conflicts. Let me know if you have any issues with the new changes before I merge this.

**Comment by marius851000** [comment]
Oh! Sorry. I tend to lose interest on stuff pretty quickly. The changes looks fine, thought.

17 avr. 2023 18:01:15 UsernameFodder ***@***.***>:

**Review by UsernameFodder — APPROVED** [review]

**Pull request merged**